plugins/crypto/ua_certificategroup_filestore.c

```diff
@@ -0,0 +1,764 @@
+    } else if(UA_NodeId_equal(&certGroup->certificateGroupId, &userTokenCertGroup)) {
+        strncpy(&rootDirectory[rootDirectorySize], "UserTokenCerts", PATH_MAX - rootDirectorySize);
+    } else {
+        UA_String nodeIdStr;
+        UA_String_init(&nodeIdStr);
+        UA_NodeId_print(&certGroup->certificateGroupId, &nodeIdStr);
+        strncpy(&rootDirectory[rootDirectorySize], (char*)nodeIdStr.data, PATH_MAX - rootDirectorySize);
+        UA_String_clear(&nodeIdStr);
+    }
+    rootDirectorySize = strnlen(rootDirectory, PATH_MAX);
+
+    context->rootFolder = UA_STRING_ALLOC(rootDirectory);
+
+    UA_StatusCode retval = UA_STATUSCODE_GOOD;
+    retval |= FileCertStore_setupStorePath("/trusted/certs", rootDirectory,
+                                           rootDirectorySize, &context->trustedCertFolder);
+    retval |= FileCertStore_setupStorePath("/trusted/crl", rootDirectory,
+                                           rootDirectorySize, &context->trustedCrlFolder);
+    retval |= FileCertStore_setupStorePath("/issuer/certs", rootDirectory,
+                                           rootDirectorySize, &context->issuerCertFolder);
+    retval |= FileCertStore_setupStorePath("/issuer/crl", rootDirectory,
+                                           rootDirectorySize, &context->issuerCrlFolder);
+    retval |= FileCertStore_setupStorePath("/rejected/certs", rootDirectory,
+                                           rootDirectorySize, &context->rejectedCertFolder);
+    retval |= FileCertStore_setupStorePath("/own/certs", rootDirectory,
+                                           rootDirectorySize, &context->ownCertFolder);
+    retval |= FileCertStore_setupStorePath("/own/private", rootDirectory,
+                                           rootDirectorySize, &context->ownKeyFolder);
+
+    return retval;
+}
+
+static UA_StatusCode
+FileCertStore_createInotifyEvent(UA_CertificateGroup *certGroup) {
+    if(certGroup == NULL) {
+        return UA_STATUSCODE_BADINTERNALERROR;
+    }
+
+    FileCertStore *context = (FileCertStore *)certGroup->context;
+
+    context->inotifyFd = inotify_init1(IN_NONBLOCK);
+    if(context->inotifyFd == -1) {
+        return UA_STATUSCODE_BADINTERNALERROR;
+    }
+
+    char rootFolder[PATH_MAX] = {0};
+    mp_snprintf(rootFolder, PATH_MAX, "%.*s",
+                (int)context->rootFolder.length, (char*)context->rootFolder.data);
+    int wd = inotify_add_watch(context->inotifyFd, rootFolder, IN_ALL_EVENTS);
+    if(wd == -1) {
+        close(context->inotifyFd);
+        context->inotifyFd = -1;
+        return UA_STATUSCODE_BADINTERNALERROR;
+    }
+
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_StatusCode
+FileCertStore_getTrustList(UA_CertificateGroup *certGroup, UA_TrustListDataType *trustList) {
+    /* Check parameter */
+    if (certGroup == NULL || trustList == NULL) {
+        return UA_STATUSCODE_BADINTERNALERROR;
+    }
```

Comment:
Dont add { } guards for a single instruction-line after an if.

Reply:
Why not? It seems to be a bad habbit of not doing it and results in cleaner (here: unambiguous) code?